evadb/binder/statement_binder.py

```diff
@@ -331,9 +337,17 @@ def _bind_func_expr(self, node: FunctionExpression):
                 )
                 # certain functions take additional inputs like yolo needs the model_name
                 # these arguments are passed by the user as part of metadata
-                node.function = lambda: function_class(
-                    **get_metadata_properties(function_obj)
-                )
+                # we also handle the special case of ChatGPT where we need to send the
+                # OpenAPI key as part of the parameter
+                # ToDO: this should be better handled
+                properties = get_metadata_properties(function_obj)
+                if node.name.upper() == str("CHATGPT"):
```

Comment:
Shall we use `string_comparison_case_insensitive` in https://github.com/georgia-tech-db/evadb/blob/staging/evadb/utils/generic_utils.py#L614 for case insensitive string comparison?